gcc/testsuite/rust/compile/complex_qualified_path_in_expr.rs

```diff
@@ -0,0 +1,6 @@
+// { dg-additional-options "-fsyntax-only" }
```

Comment:
have you tried this without -fsyntax-only? I merged in https://github.com/Rust-GCC/gccrs/pull/1954 which fixes issues there

Reply:
Good point. I tried it as per your suggestion and it exposed a couple of other issues.
1. `let _ = <<S as T1>::A as T2>::f()` failing: This was caused by `rust-hir-path.h`'s `gcc_unreachable ()` at line 948 in combination with the missing initialization for `associated_segment` in `HIR::QualifiedPathInType`'s constructor (`rust-hir-path.h` line 941) I'm guessing it wasn't intentional, right?
2. After fixing 1, `let _ = <<<S as T1>::A as T2>::B as T3>::f();` failed: This I couldn't solve completely but I'll share what I understood. `ResolveRelativeQualTypePath::resolve_qual_seg` expects a resolved node id to be returned which is not applicable if the the `Type` (as in reference to < Type as TypePath >) in the `QualifiedPathType` is a `QualifiedPathInType`. I'm not sure what the correct way of handling this is but would like to hear your input.